week1/homework/src/server.js

```diff
@@ -1,15 +1,32 @@
 'use strict';
 
 const http = require('http');
+const Util = require('./helper');
+const State = require('./fileContent');
 
 /* `createServer` MUST return an instance of `http.Server` otherwise the tests
  * will fail.
  */
-function createServer(port) {
-  let state = 10;
 
+function createServer(port) {
+  const state = new State(10);
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    if (request.url === '/state') {
```

Comment:
For a code style purpose, I would go for using a `switch` statement here instead of the `if` conditional, for readability and avoiding any kind of an errors, such as trying to send the response two times, first the actual response and the other is when you are handling the error:

so you don't have to write always:
```javascript
if (variable === '/state') {
}
if (variable === '/other-state') {
}
// in `express` this will throw an error, in normal `http` it will throw too if it sets the Headers again after they are set.
Util.handleError(response);
```

in a switch it will be as the following which make it a lot easier to read:
```javascript
switch (variable) {
  case '/state':
  case '/other-state':
  default :
}
```